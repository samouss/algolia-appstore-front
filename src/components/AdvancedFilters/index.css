.AdvancedFilters {
  display: flex;
  flex-direction: column;
  margin-top: 1vh;
}

.AdvancedFilters__Content {
  align-items: flex-start;
  margin-bottom: 1vh;
  display: none;

  @media (--tablet) {
    display: flex;
    flex-direction: row;
    margin-bottom: 0;
  }
}

.AdvancedFilters__Content--visible {
  display: flex;
  flex-direction: column;
}

.AdvancedFilters__Link {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 25px;
  color: #b9b9b9;
  font-size: 0.8rem;

  @media (--tablet) {
    display: none;
  }
}
