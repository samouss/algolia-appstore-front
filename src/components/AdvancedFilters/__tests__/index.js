import React from 'react';
import { shallow } from 'enzyme';
import AdvancedFilters from '../index';

describe('<AdvancedFilters />', () => {
  const defaultProps = {};

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AdvancedFilters
        {...props}
      >
        Content
      </AdvancedFilters>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to turn content to visible', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AdvancedFilters
        {...props}
      >
        Content
      </AdvancedFilters>,
    );

    component
      .find('[styleName="AdvancedFilters__Link"]')
      .simulate('click', {
        preventDefault: () => {},
      });

    expect(component).toMatchSnapshot();
  });

  it('expect to turn content to hide', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AdvancedFilters
        {...props}
      >
        Content
      </AdvancedFilters>,
    );

    component.setState({
      isVisible: true,
    });

    component
      .find('[styleName="AdvancedFilters__Link"]')
      .simulate('click', {
        preventDefault: () => {},
      });

    expect(component).toMatchSnapshot();
  });
});
