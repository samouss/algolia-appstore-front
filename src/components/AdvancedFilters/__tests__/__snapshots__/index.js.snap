// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedFilters /> expect to render 1`] = `
<div
  styleName="AdvancedFilters"
>
  <div
    styleName="AdvancedFilters__Content"
  >
    Content
  </div>
  <a
    href=""
    onClick={[Function]}
    styleName="AdvancedFilters__Link"
  >
    Advanced filters
    <i
      aria-hidden="true"
      className="fa fa-chevron-down"
    />
  </a>
</div>
`;

exports[`<AdvancedFilters /> expect to turn content to hide 1`] = `
<div
  styleName="AdvancedFilters"
>
  <div
    styleName="AdvancedFilters__Content"
  >
    Content
  </div>
  <a
    href=""
    onClick={[Function]}
    styleName="AdvancedFilters__Link"
  >
    Advanced filters
    <i
      aria-hidden="true"
      className="fa fa-chevron-down"
    />
  </a>
</div>
`;

exports[`<AdvancedFilters /> expect to turn content to visible 1`] = `
<div
  styleName="AdvancedFilters"
>
  <div
    styleName="AdvancedFilters__Content AdvancedFilters__Content--visible"
  >
    Content
  </div>
  <a
    href=""
    onClick={[Function]}
    styleName="AdvancedFilters__Link"
  >
    <i
      aria-hidden="true"
      className="fa fa-chevron-up"
    />
  </a>
</div>
`;
